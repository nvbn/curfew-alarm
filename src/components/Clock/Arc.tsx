import React from "react";
import { Path } from "react-native-svg";

const polarToCartesian = (
  cx: number,
  cy: number,
  r: number,
  d: number,
): [number, number] => {
  const dRadians = ((d - 90) * Math.PI) / 180;

  return [cx + r * Math.cos(dRadians), cy + r * Math.sin(dRadians)];
};

// Based on https://stackoverflow.com/a/18473154
const makeArc = (
  cx: number,
  cy: number,
  r: number,
  start: number,
  end: number,
): string => {
  const adjustedEnd = end < start ? end + 360 : end;

  const [startX, startY] = polarToCartesian(cx, cy, r, adjustedEnd); // swap?
  const [endX, endY] = polarToCartesian(cx, cy, r, start);

  const largeArcFlag = adjustedEnd - start <= 180 ? 0 : 1;

  return `
    M ${startX} ${startY}
    A ${r} ${r} 0 ${largeArcFlag} 0 ${endX} ${endY}
  `;
};

type Props = {
  startMinute: number;
  endMinute: number;
  color: string;
  width: number;
  cx: number;
  cy: number;
  r: number;
};

/**
 * SVG arc from `startMinute` to `endMinute`.
 */
const Arc = ({
  startMinute,
  endMinute,
  color,
  width,
  cx,
  cy,
  r,
}: Props): JSX.Element => (
  <Path
    d={makeArc(cx, cy, r, startMinute / 4, endMinute / 4)}
    stroke={color}
    fill="none"
    strokeWidth={width}
  />
);

export default Arc;
