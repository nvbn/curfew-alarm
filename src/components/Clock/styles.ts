import { StyleSheet } from "react-native";

export const CLOCK_WIDTH = "100%";

export const CLOCK_HEIGHT = "100%";

export const CLOCK_VIEWBOX = "0 0 100 100";

export const CLOCK_CENTER_X = 50;

export const CLOCK_CENTER_Y = 50;

export const CLOCK_MESSAGE_Y = 60;

export const CLOCK_MESSAGE_FONT_SIZE = 6;

export const CLOCK_FINE_CIRCLE_RADIUS = 45;

export const CLOCK_EMPTY_CIRCLE_RADIUS = 40;

export const CLOCK_WARNING_ARC_RADIUS = 40;

export const CLOCK_WARNING_ARC_WIDTH = 10;

export const CLOCK_CURRENT_TIME_ARC_RADIUS = 42.5;

export const CLOCK_CURRENT_TIME_ARC_WIDTH = 10;

export const CLOCK_CURRENT_TIME_TEXT_ANCHOR = "middle";

export const CLOCK_CURRENT_TIME_FONT_SIZE = 16;

export const COLOR_OK = "#33DE7C";

export const COLOR_DANGER = "#DEC028";

export const COLOR_CURFEW = "#DE5F54";

export const COLOR_CURRENT = "#3128DE";

export const COLOR_EMPTY = "#ffffff";

export const COLOR_TEXT = "#000000";

export default StyleSheet.create({
  container: {
    flex: 1,
  },
});
