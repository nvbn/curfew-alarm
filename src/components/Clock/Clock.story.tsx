import { storiesOf } from "@storybook/react-native";
import React from "react";

import Clock from "./Clock";

storiesOf("<Clock />", module)
  .add("disabled", () => (
    <Clock
      enabled={false}
      isAtHome={true}
      currentTime={{ hour: 21, minute: 0 }}
      curfewStart={{ hour: 21, minute: 0 }}
      curfewEnd={{ hour: 3, minute: 30 }}
      minutesToGoHome={30}
    />
  ))
  .add("enabled at home during normal time", () => (
    <Clock
      enabled={true}
      isAtHome={true}
      currentTime={{ hour: 11, minute: 0 }}
      curfewStart={{ hour: 21, minute: 0 }}
      curfewEnd={{ hour: 3, minute: 30 }}
      minutesToGoHome={30}
    />
  ))
  .add("enabled at home during go home time", () => (
    <Clock
      enabled={true}
      isAtHome={true}
      currentTime={{ hour: 20, minute: 45 }}
      curfewStart={{ hour: 21, minute: 0 }}
      curfewEnd={{ hour: 3, minute: 30 }}
      minutesToGoHome={30}
    />
  ))
  .add("enabled at home during curfew time", () => (
    <Clock
      enabled={true}
      isAtHome={true}
      currentTime={{ hour: 21, minute: 45 }}
      curfewStart={{ hour: 21, minute: 0 }}
      curfewEnd={{ hour: 3, minute: 30 }}
      minutesToGoHome={30}
    />
  ))
  .add("enabled outside during go home time", () => (
    <Clock
      enabled={true}
      isAtHome={false}
      currentTime={{ hour: 20, minute: 45 }}
      curfewStart={{ hour: 21, minute: 0 }}
      curfewEnd={{ hour: 3, minute: 30 }}
      minutesToGoHome={30}
    />
  ))
  .add("enabled outside during curfew time", () => (
    <Clock
      enabled={true}
      isAtHome={false}
      currentTime={{ hour: 21, minute: 45 }}
      curfewStart={{ hour: 21, minute: 0 }}
      curfewEnd={{ hour: 3, minute: 30 }}
      minutesToGoHome={30}
    />
  ))
  .add(
    "enabled outside during go home time when curfew starts after midnight",
    () => (
      <Clock
        enabled={true}
        isAtHome={false}
        currentTime={{ hour: 12, minute: 5 }}
        curfewStart={{ hour: 12, minute: 30 }}
        curfewEnd={{ hour: 1, minute: 30 }}
        minutesToGoHome={45}
      />
    ),
  )
  .add(
    "enabled outside during curfew time when curfew starts after midnight",
    () => (
      <Clock
        enabled={true}
        isAtHome={false}
        currentTime={{ hour: 12, minute: 5 }}
        curfewStart={{ hour: 12, minute: 30 }}
        curfewEnd={{ hour: 1, minute: 30 }}
        minutesToGoHome={45}
      />
    ),
  );
