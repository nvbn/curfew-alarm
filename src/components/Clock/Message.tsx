import React from "react";
import { Text } from "react-native-svg";

import i18n from "../../utils/i18n";
import {
  Status,
  STATUS_FINE,
  STATUS_GO_HOME_WHEN_CURFEW,
  STATUS_GO_HOME_WHEN_TIME_TO_GO_HOME,
  STATUS_STAY_AT_HOME_WHEN_CURFEW,
  STATUS_STAY_AT_HOME_WHEN_TIME_TO_GO_HOME,
} from "../../utils/status";
import { COLOR_CURFEW, COLOR_DANGER, COLOR_OK } from "./styles";

type BaseMessageProps = {
  text: string;
  color: string;
};

const BaseMessage = ({ text, color }: BaseMessageProps): JSX.Element => (
  <Text x={50} y={60} fill={color} textAnchor="middle" fontSize={6}>
    {text}
  </Text>
);

type Props = {
  status: Status;
};

/**
 * A message in SVG with the action expected from a user.
 */
const Message = ({ status }: Props): JSX.Element => {
  switch (status) {
    case STATUS_FINE:
      return <BaseMessage text={i18n.t("clockMessageFine")} color={COLOR_OK} />;
    case STATUS_GO_HOME_WHEN_TIME_TO_GO_HOME:
      return (
        <BaseMessage
          text={i18n.t("clockMessageTimeToGoHome")}
          color={COLOR_DANGER}
        />
      );
    case STATUS_STAY_AT_HOME_WHEN_TIME_TO_GO_HOME:
      return (
        <BaseMessage
          text={i18n.t("clockMessageStayAtHome")}
          color={COLOR_DANGER}
        />
      );
    case STATUS_GO_HOME_WHEN_CURFEW:
      return (
        <BaseMessage text={i18n.t("clockMessageGoHome")} color={COLOR_CURFEW} />
      );
    case STATUS_STAY_AT_HOME_WHEN_CURFEW:
      return (
        <BaseMessage
          text={i18n.t("clockMessageStayAtHome")}
          color={COLOR_CURFEW}
        />
      );
  }
};

export default Message;
