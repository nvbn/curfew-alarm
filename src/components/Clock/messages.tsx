import React from "react";
import { Text } from "react-native-svg";

import {
  Status,
  STATUS_FINE,
  STATUS_GO_HOME_WHEN_CURFEW,
  STATUS_GO_HOME_WHEN_TIME_TO_GO_HOME,
  STATUS_STAY_AT_HOME_WHEN_CURFEW,
  STATUS_STAY_AT_HOME_WHEN_TIME_TO_GO_HOME,
} from "../../utils/status";
import { COLOR_CURFEW, COLOR_DANGER, COLOR_OK } from "./styles";

type BaseMessageProps = {
  text: string;
  color: string;
};

const BaseMessage = ({ text, color }: BaseMessageProps): JSX.Element => (
  <Text x={50} y={60} fill={color} textAnchor="middle" fontSize={6}>
    {text}
  </Text>
);

type Props = {
  status: Status;
};

const Message = ({ status }: Props): JSX.Element => {
  switch (status) {
    case STATUS_FINE:
      return <BaseMessage text="You're fine" color={COLOR_OK} />;
    case STATUS_GO_HOME_WHEN_TIME_TO_GO_HOME:
      return <BaseMessage text="Time to go home" color={COLOR_DANGER} />;
    case STATUS_STAY_AT_HOME_WHEN_TIME_TO_GO_HOME:
      return <BaseMessage text="Stay at home" color={COLOR_DANGER} />;
    case STATUS_GO_HOME_WHEN_CURFEW:
      return <BaseMessage text="Go home!" color={COLOR_CURFEW} />;
    case STATUS_STAY_AT_HOME_WHEN_CURFEW:
      return <BaseMessage text="Stay at home" color={COLOR_CURFEW} />;
  }
};

export default Message;
