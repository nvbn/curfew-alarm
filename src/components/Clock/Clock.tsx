import React from "react";
import Svg, { Circle, Text } from "react-native-svg";

import { getStatus } from "../../utils/status";
import { formatTime, Time, toMinutes } from "../../utils/time";
import Arc from "./Arc";
import Message from "./Message";
import styles, {
  CLOCK_CENTER_X,
  CLOCK_CENTER_Y,
  CLOCK_CURRENT_TIME_ARC_RADIUS,
  CLOCK_CURRENT_TIME_ARC_WIDTH,
  CLOCK_CURRENT_TIME_FONT_SIZE,
  CLOCK_CURRENT_TIME_TEXT_ANCHOR,
  CLOCK_EMPTY_CIRCLE_RADIUS,
  CLOCK_FINE_CIRCLE_RADIUS,
  CLOCK_HEIGHT,
  CLOCK_VIEWBOX,
  CLOCK_WARNING_ARC_RADIUS,
  CLOCK_WARNING_ARC_WIDTH,
  CLOCK_WIDTH,
  COLOR_CURFEW,
  COLOR_CURRENT,
  COLOR_DANGER,
  COLOR_EMPTY,
  COLOR_OK,
  COLOR_TEXT,
} from "./styles";

type Props = {
  enabled: boolean;
  isAtHome: boolean;
  currentTime: Time;

  curfewStart: Time;
  curfewEnd: Time;

  minutesToGoHome: number;
};

type WarningsArcsProps = {
  curfewStart: Time;
  curfewEnd: Time;
  minutesToGoHome: number;
};

const WarningArcs = ({
  curfewStart,
  curfewEnd,
  minutesToGoHome,
}: WarningsArcsProps): JSX.Element => (
  <>
    <Arc
      cx={CLOCK_CENTER_X}
      cy={CLOCK_CENTER_Y}
      r={CLOCK_WARNING_ARC_RADIUS}
      width={CLOCK_WARNING_ARC_WIDTH}
      color={COLOR_CURFEW}
      startMinute={toMinutes(curfewStart)}
      endMinute={toMinutes(curfewEnd)}
    />
    <Arc
      cx={CLOCK_CENTER_X}
      cy={CLOCK_CENTER_Y}
      r={CLOCK_WARNING_ARC_RADIUS}
      width={CLOCK_WARNING_ARC_WIDTH}
      color={COLOR_DANGER}
      startMinute={toMinutes(curfewStart) - minutesToGoHome}
      endMinute={toMinutes(curfewStart)}
    />
  </>
);

/**
 * Clock that displays time, indicates when the curfew starts,
 * indicates when to go home, and shows a message with expected action.
 */
const Clock = ({
  enabled,
  isAtHome,
  currentTime,
  curfewStart,
  curfewEnd,
  minutesToGoHome,
}: Props): JSX.Element => (
  <Svg
    width={CLOCK_WIDTH}
    height={CLOCK_HEIGHT}
    viewBox={CLOCK_VIEWBOX}
    style={styles.container}
  >
    <Circle
      cx={CLOCK_CENTER_X}
      cy={CLOCK_CENTER_Y}
      r={CLOCK_FINE_CIRCLE_RADIUS}
      fill={COLOR_OK}
    />
    {enabled && (
      <WarningArcs
        curfewStart={curfewStart}
        curfewEnd={curfewEnd}
        minutesToGoHome={minutesToGoHome}
      />
    )}
    <Circle
      cx={CLOCK_CENTER_X}
      cy={CLOCK_CENTER_Y}
      r={CLOCK_EMPTY_CIRCLE_RADIUS}
      fill={COLOR_EMPTY}
    />
    <Arc
      cx={CLOCK_CENTER_X}
      cy={CLOCK_CENTER_Y}
      r={CLOCK_CURRENT_TIME_ARC_RADIUS}
      width={CLOCK_CURRENT_TIME_ARC_WIDTH}
      color={COLOR_CURRENT}
      startMinute={toMinutes(currentTime) - 1}
      endMinute={toMinutes(currentTime) + 1}
    />
    <Text
      x={CLOCK_CENTER_X}
      y={CLOCK_CENTER_Y}
      fill={COLOR_TEXT}
      textAnchor={CLOCK_CURRENT_TIME_TEXT_ANCHOR}
      fontSize={CLOCK_CURRENT_TIME_FONT_SIZE}
    >
      {formatTime(currentTime)}
    </Text>
    <Message
      status={getStatus(
        enabled,
        isAtHome,
        currentTime,
        curfewStart,
        curfewEnd,
        minutesToGoHome,
      )}
    />
  </Svg>
);

export default Clock;
