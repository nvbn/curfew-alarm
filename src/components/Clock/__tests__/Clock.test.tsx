import { render } from "@testing-library/react-native";
import React from "react";

import Clock from "../Clock";

describe("<Clock />", () => {
  test("during the regular time", () => {
    const rendered = render(
      <Clock
        enabled={true}
        isAtHome={false}
        currentTime={{ hour: 12, minute: 30 }}
        curfewStart={{ hour: 21, minute: 0 }}
        curfewEnd={{ hour: 4, minute: 30 }}
        minutesToGoHome={15}
      />,
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });

  test("when it's time to go home", () => {
    const rendered = render(
      <Clock
        enabled={true}
        isAtHome={false}
        currentTime={{ hour: 20, minute: 50 }}
        curfewStart={{ hour: 21, minute: 0 }}
        curfewEnd={{ hour: 4, minute: 30 }}
        minutesToGoHome={15}
      />,
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });

  test("when curfew is active", () => {
    const rendered = render(
      <Clock
        enabled={true}
        isAtHome={false}
        currentTime={{ hour: 1, minute: 0 }}
        curfewStart={{ hour: 21, minute: 0 }}
        curfewEnd={{ hour: 4, minute: 30 }}
        minutesToGoHome={15}
      />,
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });

  test("when disabled", () => {
    const rendered = render(
      <Clock
        enabled={false}
        isAtHome={false}
        currentTime={{ hour: 1, minute: 0 }}
        curfewStart={{ hour: 21, minute: 0 }}
        curfewEnd={{ hour: 4, minute: 30 }}
        minutesToGoHome={15}
      />,
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
