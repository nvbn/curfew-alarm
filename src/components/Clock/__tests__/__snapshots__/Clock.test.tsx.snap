// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> during the regular time 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="100%"
  bbWidth="100%"
  focusable={false}
  height="100%"
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 1,
      },
      Object {
        "flex": 0,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  vbHeight={100}
  vbWidth={100}
  width="100%"
>
  <RNSVGGroup>
    <RNSVGCircle
      cx={50}
      cy={50}
      fill={4281589372}
      propList={
        Array [
          "fill",
        ]
      }
      r={45}
    />
    <RNSVGPath
      d="
    M 86.95518130045147 34.69266270539641
    A 40 40 0 0 0 21.71572875253809 21.7157287525381
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4292763476}
      strokeWidth={10}
    />
    <RNSVGPath
      d="
    M 21.71572875253809 21.7157287525381
    A 40 40 0 0 0 19.926407700840905 23.626167395997246
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4292788264}
      strokeWidth={10}
    />
    <RNSVGCircle
      cx={50}
      cy={50}
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
      r={40}
    />
    <RNSVGPath
      d="
    M 44.26883546336677 92.1118006389425
    A 42.5 42.5 0 0 0 44.636543811727236 92.1602103613879
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4281411806}
      strokeWidth={10}
    />
    <RNSVGText
      content={null}
      dx={Array []}
      dy={Array []}
      fill={4278190080}
      font={
        Object {
          "fontSize": 16,
          "textAnchor": "middle",
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      rotate={Array []}
      x={
        Array [
          50,
        ]
      }
      y={
        Array [
          50,
        ]
      }
    >
      <RNSVGTSpan
        content="12:30"
        dx={Array []}
        dy={Array []}
        font={Object {}}
        rotate={Array []}
        x={Array []}
        y={Array []}
      />
    </RNSVGText>
    <RNSVGText
      content={null}
      dx={Array []}
      dy={Array []}
      fill={4281589372}
      font={
        Object {
          "fontSize": 6,
          "textAnchor": "middle",
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      rotate={Array []}
      x={
        Array [
          50,
        ]
      }
      y={
        Array [
          60,
        ]
      }
    >
      <RNSVGTSpan
        content="You're fine"
        dx={Array []}
        dy={Array []}
        font={Object {}}
        rotate={Array []}
        x={Array []}
        y={Array []}
      />
    </RNSVGText>
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`<Clock /> when curfew is active 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="100%"
  bbWidth="100%"
  focusable={false}
  height="100%"
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 1,
      },
      Object {
        "flex": 0,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  vbHeight={100}
  vbWidth={100}
  width="100%"
>
  <RNSVGGroup>
    <RNSVGCircle
      cx={50}
      cy={50}
      fill={4281589372}
      propList={
        Array [
          "fill",
        ]
      }
      r={45}
    />
    <RNSVGPath
      d="
    M 86.95518130045147 34.69266270539641
    A 40 40 0 0 0 21.71572875253809 21.7157287525381
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4292763476}
      strokeWidth={10}
    />
    <RNSVGPath
      d="
    M 21.71572875253809 21.7157287525381
    A 40 40 0 0 0 19.926407700840905 23.626167395997246
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4292788264}
      strokeWidth={10}
    />
    <RNSVGCircle
      cx={50}
      cy={50}
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
      r={40}
    />
    <RNSVGPath
      d="
    M 61.17882661446394 8.99653873727545
    A 42.5 42.5 0 0 0 60.820582798734925 8.900547596157416
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4281411806}
      strokeWidth={10}
    />
    <RNSVGText
      content={null}
      dx={Array []}
      dy={Array []}
      fill={4278190080}
      font={
        Object {
          "fontSize": 16,
          "textAnchor": "middle",
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      rotate={Array []}
      x={
        Array [
          50,
        ]
      }
      y={
        Array [
          50,
        ]
      }
    >
      <RNSVGTSpan
        content="01:00"
        dx={Array []}
        dy={Array []}
        font={Object {}}
        rotate={Array []}
        x={Array []}
        y={Array []}
      />
    </RNSVGText>
    <RNSVGText
      content={null}
      dx={Array []}
      dy={Array []}
      fill={4281589372}
      font={
        Object {
          "fontSize": 6,
          "textAnchor": "middle",
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      rotate={Array []}
      x={
        Array [
          50,
        ]
      }
      y={
        Array [
          60,
        ]
      }
    >
      <RNSVGTSpan
        content="You're fine"
        dx={Array []}
        dy={Array []}
        font={Object {}}
        rotate={Array []}
        x={Array []}
        y={Array []}
      />
    </RNSVGText>
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`<Clock /> when it's time to go home 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="100%"
  bbWidth="100%"
  focusable={false}
  height="100%"
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 1,
      },
      Object {
        "flex": 0,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  vbHeight={100}
  vbWidth={100}
  width="100%"
>
  <RNSVGGroup>
    <RNSVGCircle
      cx={50}
      cy={50}
      fill={4281589372}
      propList={
        Array [
          "fill",
        ]
      }
      r={45}
    />
    <RNSVGPath
      d="
    M 86.95518130045147 34.69266270539641
    A 40 40 0 0 0 21.71572875253809 21.7157287525381
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4292763476}
      strokeWidth={10}
    />
    <RNSVGPath
      d="
    M 21.71572875253809 21.7157287525381
    A 40 40 0 0 0 19.926407700840905 23.626167395997246
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4292788264}
      strokeWidth={10}
    />
    <RNSVGCircle
      cx={50}
      cy={50}
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
      r={40}
    />
    <RNSVGPath
      d="
    M 18.791293348983363 21.150968314849973
    A 42.5 42.5 0 0 0 18.54072958180963 21.42441068402662
  "
      fill={null}
      propList={
        Array [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={4281411806}
      strokeWidth={10}
    />
    <RNSVGText
      content={null}
      dx={Array []}
      dy={Array []}
      fill={4278190080}
      font={
        Object {
          "fontSize": 16,
          "textAnchor": "middle",
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      rotate={Array []}
      x={
        Array [
          50,
        ]
      }
      y={
        Array [
          50,
        ]
      }
    >
      <RNSVGTSpan
        content="20:50"
        dx={Array []}
        dy={Array []}
        font={Object {}}
        rotate={Array []}
        x={Array []}
        y={Array []}
      />
    </RNSVGText>
    <RNSVGText
      content={null}
      dx={Array []}
      dy={Array []}
      fill={4292788264}
      font={
        Object {
          "fontSize": 6,
          "textAnchor": "middle",
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      rotate={Array []}
      x={
        Array [
          50,
        ]
      }
      y={
        Array [
          60,
        ]
      }
    >
      <RNSVGTSpan
        content="Time to go home"
        dx={Array []}
        dy={Array []}
        font={Object {}}
        rotate={Array []}
        x={Array []}
        y={Array []}
      />
    </RNSVGText>
  </RNSVGGroup>
</RNSVGSvgView>
`;
